tests/testthat/test-subset_obs.R

```diff
@@ -0,0 +1,93 @@
+# Daily data
+obs_daily <- enw_example("nowcast")$latest[[1]]
+# Mock weekly data
+obs_weekly <- data.table(
+    reference_date = as.IDate("2021-07-28") + 7 * (0:6),
+    report_date = c(as.IDate("2021-08-25") + 7 * (0:2),
+                    rep(as.IDate("2021-09-08"), 4)),
+    .group = rep(1, times = 7),
+    max_confirm = c(625, 856, 1073, 1733, 2268, 2388, 1487),
+    confirm = c(615, 847, 1073, 1733, 2268, 2388, 1487),
+    cum_prop_reported = c(615 / 625, 847 / 856, rep(1, 5)),
+    delay = c(rep(4, 3), 3:0)
+)
+
+# Test 1: Select "after" (daily)
+test_that("select 'after' returns correct subset of reference dates when timesteps are days", { # nolint
+  max_delay <- 5
+  internal_timestep <- 1
+  expected_cutoff <- max(obs_daily$reference_date) -
+    max_delay * internal_timestep
+  result <- subset_obs(obs_daily, max_delay, internal_timestep,
+                       select = "after")
+  expect_true(all(result$reference_date > expected_cutoff))
+  expect_identical(unique(diff(result$reference_date)),
+                   as.integer(internal_timestep))
+})
+
+# Test 2: Select "before" (daily)
+test_that("select 'before' returns correct subset of reference dates when timesteps are days", { # nolint
```

Comment:
I usually put the function name at the beginning of the test string. - not sure if it really matters but may as well follow that pattern unless a reason not to?